### ISSUE TITLE
New ProductRepositoryInterfaceV2 breaks Magento WebAPI functionality for swagger

### ISSUE THREAD

**Issue by zack6849** [springbot/magento2-plugin#35]
Hi There!

It seems your new class introduced in commit https://github.com/springbot/magento2-plugin/commit/777cc3e79a3d73af00fde0543aae46198ed6cf6f#diff-7ecff4ab7ab3441d131e4334a05736ad breaks magento 2's swagger interface with an error because it expects the string to end with "Interface" and yours is InterfaceV2, making the code in Magento\\WebApi\\Model\\ServiceMetadata::getServiceName() fail as there are only 3 matches, not 4.

![screen shot 2018-09-19 at 10 33 26 am](https://user-images.githubusercontent.com/1343898/45760180-7f307b00-bbf7-11e8-9cac-fee594fdfebe.png)

Resulting in the following stack trace:

```
Notice: Undefined offset: 4 in /var/www/html/vendor/magento/module-webapi/Model/ServiceMetadata.php on line 216

#0 /var/www/html/vendor/magento/module-webapi/Model/ServiceMetadata.php(216): Magento\Framework\App\ErrorHandler-&gt;handler(8, 'Undefined offse...', '/var/www/html/v...', 216, Array)
#1 /var/www/html/vendor/magento/module-webapi/Model/ServiceMetadata.php(115): Magento\Webapi\Model\ServiceMetadata-&gt;getServiceName('Springbot\\Main\\...', 'V2')
#2 /var/www/html/vendor/magento/module-webapi/Model/ServiceMetadata.php(156): Magento\Webapi\Model\ServiceMetadata-&gt;initServicesMetadata()
#3 /var/www/html/vendor/magento/module-webapi/Model/Rest/Swagger/Generator.php(923): Magento\Webapi\Model\ServiceMetadata-&gt;getServicesConfig()
#4 /var/www/html/vendor/magento/module-webapi/Controller/Rest/SchemaRequestProcessor.php(53): Magento\Webapi\Model\Rest\Swagger\Generator-&gt;getListOfServices()
#5 /var/www/html/vendor/magento/module-webapi/Controller/Rest.php(244): Magento\Webapi\Controller\Rest\SchemaRequestProcessor-&gt;process(Object(Magento\Framework\Webapi\Rest\Request\Proxy))
#6 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Webapi\Controller\Rest-&gt;dispatch(Object(Magento\Framework\App\Request\Http))
#7 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Webapi\Controller\Rest\Interceptor-&gt;___callParent('dispatch', Array)
#8 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Webapi\Controller\Rest\Interceptor-&gt;Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#9 /var/www/html/generated/code/Magento/Webapi/Controller/Rest/Interceptor.php(26): Magento\Webapi\Controller\Rest\Interceptor-&gt;___callPlugins('dispatch', Array, Array)
#10 /var/www/html/vendor/magento/framework/App/Http.php(135): Magento\Webapi\Controller\Rest\Interceptor-&gt;dispatch(Object(Magento\Framework\App\Request\Http))
#11 /var/www/html/vendor/magento/framework/App/Bootstrap.php(257): Magento\Framework\App\Http-&gt;launch()
#12 /var/www/html/index.php(39): Magento\Framework\App\Bootstrap-&gt;run(Object(Magento\Framework\App\Http\Interceptor))
#13 {main}
```

**Comment by JoeyK4816** [comment]
Thank you for submitting this issue. Our team will review your changes internally and respond here.

**Comment by JoeyK4816** [comment]
@zack6849 
Thanks for your report. This issue is being addressed in the pull request #37. We are currently testing the modification but had to make more updates than what was in your pull request. This PR will be merged into release version 1.6.12 later today.
